tfs/src/main/java/hudson/plugins/tfs/TeamBuildEndpoint.java

```diff
@@ -54,6 +56,7 @@
     private static final Map<String, AbstractCommand.Factory> COMMAND_FACTORIES_BY_NAME;
     public static final String URL_NAME = "team-build";
     public static final String PARAMETER = "parameter";
+    public static final String BUILD_SOURCE_BRANCH = "Build.SourceBranch";

```

Comment:
Great call.  Made changes to get the branch from:
1. jobName input.
2. Build.SourceBranch variable

Throw if none of them are available.